SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -799,7 +807,8 @@ FNetworkGUID FSpatialNetGUIDCache::GetOrAssignNetGUID_SpatialGDK(UObject* Object
 			   *Cast<USpatialNetDriver>(Driver)->Connection->GetWorkerId(), *Object->GetPathName(), *NetGUID.ToString());
 	}
 
-	check((NetGUID.IsValid() && !NetGUID.IsDefault()) || Object == nullptr);
+	checkf((NetGUID.IsValid() && !NetGUID.IsDefault()) || Object == nullptr, TEXT("NetGUID: %s, Object %s"), *NetGUID.ToString(),
```

Comment:
The message you've added doesn't indicate *why* the check failed - worth adding some context.

Reply:
Yeah I've only extended the message to see what NetGUID value we had - unsure on how to improve it since the check condition is complex: `Either NetGUID is invalid or default, or the object is None: NetGUID: %s, Object: %s` doesn't seem to help 😆

Reply:
I would recommend splitting this into multiple checks for clarity.

Reply:
Done.